frontend/src/share/DialogWindows/_dialogWindows/ShowStudentsOnGroupDialog.js

```diff
@@ -0,0 +1,204 @@
+import React, { useEffect, useState } from 'react';
+
+import PropTypes from 'prop-types';
+
+import Button from '@material-ui/core/Button';
+
+import { useTranslation } from 'react-i18next';
+import { isEmpty } from 'lodash';
+import i18n from '../../../helper/i18n';
+import RenderStudentTable from '../../../helper/renderStudentTable';
+import { getAllStudentsByGroupId } from '../../../services/studentService';
+import { UploadFile } from '../../../components/UploadFile/UploadFile';
+import CustomDialog from '../CustomDialog';
+import MovingGroupsDialog from './MovingGroupsDialog';
+
+const ShowStudentsOnGroupDialog = (props) => {
+    const { onClose, cardId, open, onDeleteStudent, students, onSubmit, match, groups, group } =
+        props;
+    const [checkBoxStudents, setCheckBoxStudents] = useState([]);
+    const [isGroupButtonDisabled, setIsGroupButtonDisabled] = useState(true);
+    const [checkedAll, setCheckedAll] = useState(false);
+    const [openUploadFile, setOpenUploadFile] = useState(false);
+    const [showStudentList, setShowStudentList] = useState(false);
+    const { t } = useTranslation('formElements');
+
+    const setSelectDisabled = () => {
+        checkBoxStudents.forEach((item) => {
+            if (item.checked) {
+                setIsGroupButtonDisabled(false);
+            }
+        });
+    };
+    const parseStudentToCheckBox = () => {
+        const res = students.map((item) => {
+            return { ...item, checked: false };
+        });
+        setCheckBoxStudents(res);
+    };
+    useEffect(() => {
+        getAllStudentsByGroupId(props.group.id);
+    }, [open, openUploadFile]);
+    useEffect(() => {
+        parseStudentToCheckBox();
+    }, [props.students]);
+    useEffect(() => {
+        setSelectDisabled();
+    }, [props.students]);
+
+    const handleClose = () => {
+        onClose(cardId);
+    };
+
+    const handleCheckElement = (event) => {
+        setCheckBoxStudents(
+            checkBoxStudents.map((item, index) => {
+                if (item.id === Number(event.target.value)) {
+                    checkBoxStudents[index].checked = event.target.checked;
```

Comment:
`checkBoxStudents[index]` is `item`

Reply:
checkBoxStudents[index] so we avoid no-param-reassign